File: src/83_lesson/ToDo/src/modules/todos/models/ToDoItem.ts
import type { Guid } from "guid-typescript";
import type { IMappable } from "@/infrastructure/mappers/IMappable";

export class ToDoItem implements IMappable<ToDoItem>{

    constructor() {
        this.dueTime = new Date();
        this.reminderTime = new Date();
    }

    id!: Guid;
    title!: string;
    notes!: string;
    isDone!: boolean;
    isFavourite!: boolean;
    dueTime!: Date;
    reminderTime!: Date;

    mapFrom(object: any): ToDoItem {
        this.id = object.id;
        this.title = object.title;
        this.notes = object.notes;
        this.isDone = object.isDone;
        this.isFavourite = object.isFavourite;
        this.dueTime = new Date(object.dueTime);
        this.reminderTime = new Date(object.reminderTime);

        return this;
    }

}

File: src/83_lesson/Todo.ServerApp/Todo.Domain/Entities/TodoItem.cs
using Todo.Domain.Common.Entities;

namespace Todo.Domain.Entities;

public class TodoItem : AuditableEntity
{
    public string Title { get; set; } = default!;

    public bool IsDone { get; set; }

    public bool IsFavorite { get; set; }

    public DateTimeOffset DueTime { get; set; }

    public DateTimeOffset ReminderTime { get; set; }
}